tests/bugs/gh-3931.slang

```diff
@@ -0,0 +1,24 @@
+//TEST:SIMPLE(filecheck=CHECK): -O0 -target spirv -emit-spirv-directly -stage compute -entry computeMain -matrix-layout-row-major
+//COM:TEST:SIMPLE(filecheck=CHECK): -O2 -target spirv -emit-spirv-directly -stage compute -entry computeMain -matrix-layout-row-major
```

Comment:
Is the second test meant to be commented out with "//COM:"?
I thought this test is to test if the row-major-ness works properly or not when "-O2" is used, isn't it?

Reply:
there were 2 problems:
> Incorrect slangc arg parsing (possible to have row major & column major in option_set) -- addressed in pr.

> spirv-opt 2023-6 & upstream incorrectly optimizes code and removes member decorations in some odd cases, works with spirv-opt 2024-1 -- not solved in this pr since we use upstream of spirv-opt currently to solve different issues.

Of these, we can only fix problem one since this is a problem inside Slang. This allows all code to work every-run with row major'ness set.
Problem 2 cannot be addressed since the spirv-opt version we are using cannot be downgraded. using O0 implies no optimization, O2 implies using spirv-opt, which we cannot do.